src/hints/mod.rs

```diff
@@ -336,3 +339,24 @@ pub fn breakpoint(
     println!("\n-----------END BREAKPOINT-----------\n");
     Ok(())
 }
+
+const IS_ON_CURVE: &str = "ids.is_on_curve = (y * y) % SECP_P == y_square_int";
+pub fn is_on_curve(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let y: BigInt = exec_scopes.get("y")?;
+    let y_square_int: BigInt = exec_scopes.get("y_square_int")?;
+    // TODO: assume SECP_P is in scope, or should we use this constant?
+    // let SECP_P = cairo_vm::hint_processor::builtin_hint_processor::secp::secp_utils::SECP_P;
+    let SECP_P: BigInt = exec_scopes.get("SECP_P")?;
+
+    let is_on_curve = (y.clone() * y) % SECP_P == y_square_int;
+    let is_on_curve: Felt252 = if is_on_curve { Felt252::from(1) } else { Felt252::from(0) };
```

Comment:
Please review: is this the right way to handle booleans in a felt-only world?

Reply:
That's fine by me, you can also use `Felt252::ONE` and `Felt252::ZERO` like Herman did for consistency.

Reply:
ca2677b